docs/topics/gradle.md

```diff
@@ -968,3 +968,74 @@ When configuring the Kotlin daemon's JVM arguments, note that:
   >
   {type="note"}
 * If the `Xmx` is not specified, the Kotlin daemon will inherit it from the Gradle daemon.
+
+## Defining Kotlin compiler execution strategy
+
+_Kotlin compiler execution strategy_ describes where Kotlin compiler is executed and if incremental compilation is supported in each case.
+
+There are three compiler execution strategies:
+
+| Strategy       | Where Kotlin compiler is executed | Support for incremental compilation | Other characteristics                                                  |
+|----------------|-----------------------------------|-------------------------------------|------------------------------------------------------------------------|
+| Daemon         | Inside its own daemon process     | Yes                                 | *The default strategy*. Can be shared between different Gradle daemons |
+| In process     | Inside the Gradle daemon process  | No                                  | May share heap with the Gradle daemon                                  |
+| Out of process | In a separate forked process      | No                                  | -                                                                      |
+
+To define a Kotlin compiler execution strategy, you can use one of the following properties:
+* The `kotlin.compiler.execution.strategy` Gradle property.
+* The `compilerExecutionStrategy` compile task property.
+* The `-Dkotlin.compiler.execution.strategy` system property, which will be removed in future releases. 
```

Comment:
Let's put the last thought in a separate note. Also, is it deprecated in 1.6.20? If yes, say this explicitly.

Reply:
>is it deprecated in 1.6.20?

I don't have such info, @ALikhachev could you help, please?

Reply:
It's not deprecated yet, and we don't show any warnings on it's usages. We will add deprecation warning in 1.7 and then remove it in 1.8.